tests/integration/base_block_test.go

```diff
@@ -143,7 +144,13 @@ func runBlockE2ETestLite(helper *clients.TestClient, k8sh *utils.K8sHelper, s su
 func checkPoolDeleted(helper *clients.TestClient, s suite.Suite, name string) {
 	i := 0
 	for i < utils.RetryLoop {
-		if !foundPool(helper, s, name) {
+		found, err := foundPool(helper, s, name)
+		if err != nil {
+			// try again on failure since the pool may have been in an unexpected state while deleting
+			logger.Warningf("error getting pools. %+v", err)
+			continue
```

Comment:
is a small sleep in between retries needed?

Reply:
yes, definitely, good catch